src/Compilers/CSharp/Test/Semantic/Semantics/ScriptSemanticsTests.cs

```diff
@@ -16,6 +16,155 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+        public System.Threading.Tasks.Task<T> Task => null;
+        public void SetException(Exception e) { }
+        public void SetResult(T result) { }
+        public void AwaitOnCompleted<A, B>(ref A a, ref B b) { }
+        public void AwaitUnsafeOnCompleted<A, B>(ref A a, ref B b) { }
+        public void Start<A>(ref A a) { }
+    }
+}
+
+namespace System.Runtime.CompilerServices {
+    public class ExtensionAttribute {}
+}
+
+namespace System {
+    public delegate void Action();
+}
+namespace System.Runtime.CompilerServices {
+    public interface INotifyCompletion {
+        void OnCompleted(System.Action action);
+    }
+}
+
+namespace System.Threading.Tasks {
+    public class Awaiter<T>: System.Runtime.CompilerServices.INotifyCompletion {
+        public bool IsCompleted  => true;
+        public void OnCompleted(System.Action action) {}
+        public T GetResult() { throw new Exception(); }
+    }
+    public class Task<T> {}
+}
+namespace Hidden {
+public static class MyExtensions {
+    public static System.Threading.Tasks.Awaiter<T> GetAwaiter<T>(this System.Threading.Tasks.Task<T> task) {
+        return null;
+    }
+}
+}";
+
+            var corCompilation = CreateCompilation(corAssembly, references: new[] { MscorlibRef_v20 });
+            corCompilation.VerifyDiagnostics();
+            return corCompilation.ToMetadataReference();
+        }
+
+        [WorkItem(19048, "https://github.com/dotnet/roslyn/pull/19623")]
+        [Fact]
+        public void NonStandardTaskImplementation_NoGlobalUsing_NoScriptUsing()
+        {
+
+            var script = CreateCompilation(
+                source: @" System.Console.Write(""complete"");",
+                parseOptions: TestOptions.Script,
+                options: TestOptions.DebugExe,
+                references: new MetadataReference[] { TaskFacadeAssembly(), MscorlibRef_v20 });
+
+            script.VerifyEmitDiagnostics(
+                // error CS1061: 'Task<object>' does not contain a definition for 'GetAwaiter' and no extension method 'GetAwaiter' accepting a first argument of type 'Task<object>' could be found (are you missing a using directive or an assembly reference?)
+                Diagnostic(ErrorCode.ERR_NoSuchMemberOrExtension, "").WithArguments("System.Threading.Tasks.Task<object>", "GetAwaiter").WithLocation(1, 1));
+
+        }
+
+        [WorkItem(19048, "https://github.com/dotnet/roslyn/pull/19623")]
+        [Fact]
+        public void NonStandardTaskImplementation_GlobalUsing_NoScriptUsing()
+        {
+            var script = CreateCompilation(
+                source: @" System.Console.Write(""complete"");",
+                parseOptions: TestOptions.Script,
+                options: TestOptions.DebugExe.WithUsings("Hidden"),
+                references: new MetadataReference[] { TaskFacadeAssembly(), MscorlibRef_v20 });
+
+            script.VerifyEmitDiagnostics();
+
+            var compiled = CompileAndVerify(script);
```

Comment:
Never mind, our Task type doesn't have real implementation.

---
In reply to: [119920331](https://github.com/dotnet/roslyn/pull/19623#discussion_r119920331) [](ancestors = 119920331)